### ISSUE TITLE
Terraform fails if provided RDS parameter is static.

### ISSUE THREAD

**Issue by 770715** [terraform-aws-modules/terraform-aws-rds#349]
## Description

If you want to create a parameters group and provide a static parameter (ApplyType), Terraform fails.

## Versions

 - Terraform v1.0.1

 - Provider(s): AWS c3.48.0

## Reproduction
Try to create an RDS instance + parameter groups and use a static parameter (e.g. query_cache_type for MySQL).
Example based on the module docs:

```hcl
module "db" {
  source  = "terraform-aws-modules/rds/aws"
  version = "~> 3.0"

  identifier = "demodb"

  engine            = "mysql"
  engine_version    = "5.7.19"
  instance_class    = "db.t2.large"
  allocated_storage = 5

  name     = "demodb"
  username = "user"
  password = "YourPwdShouldBeLongAndSecure!"
  port     = "3306"

  iam_database_authentication_enabled = true

  vpc_security_group_ids = ["sg-12345678"]

  maintenance_window = "Mon:00:00-Mon:03:00"
  backup_window      = "03:00-06:00"

  # Enhanced Monitoring - see example for details on how to create the role
  # by yourself, in case you don't want to create it automatically
  monitoring_interval = "30"
  monitoring_role_name = "MyRDSMonitoringRole"
  create_monitoring_role = true

  tags = {
    Owner       = "user"
    Environment = "dev"
  }

  # DB subnet group
  subnet_ids = ["subnet-12345678", "subnet-87654321"]

  # DB parameter group
  family = "mysql5.7"

  # DB option group
  major_engine_version = "5.7"

  # Database Deletion Protection
  deletion_protection = true

  parameters = [
    {
      name = "innodb_purge_threads"
      value = "1"
    },
    {
      name = "skip_name_resolve"
      value = "1"
    }
  ]
}
```

## Expected behavior
Terraform should build an instance + a custom parameters group.

## Actual behavior
Terraform fails with the following error:

```
Error: Error modifying DB Parameter Group: InvalidParameterCombination: cannot use immediate apply method for static parameter
	status code: 400, request id: e7da6cb3-5fac-4bts-863c-pp833efa7751
```

## Additional context
At the moment, the module responsible for creating parameter groups forces i"mmediate apply" method, and there is no way to overwrite this.

**Comment by bryantbiggs** [comment]
hi @770715 do you have an example? without any example its quite hard to follow what you are describing

**Comment by 770715** [comment]
Hi. I updated the issue description with a sample code. There are two parameters there:

```hcl
 {
      name = "innodb_purge_threads"
      value = "1"
    },
    {
      name = "skip_name_resolve"
      value = "1"
}
```

Both parameters are static, so they can not be set without rebooting the instance. Problem is, the module creating custom parameters groups applies `apply_method = "immediate"` (seems to be def Terraform behaviour) to all parameters. I checked default AWS parameters groups (e.g. `aws rds describe-db-parameters --db-parameter-group-name "default.mysql8.0"`) for a few engines, and they apply `"ApplyMethod": "pending-reboot"` to all parameters.

Thanks for your time.

**Comment by mshah07** [comment]
Hi, Having the same issue here. 
For example: This is the parameter map I am passing
```
      parameters = [
    {
      name  = "rds.logical_replication"
      value = 1
    }
  ]
```
The output is :
```
parameter {
      + apply_method = "immediate"
      + name         = "rds.logical_replication"
      + value        = "1"
    }
```
rds.logical_replication is a static parameter and cannot be modified during runtime, it needs DB reboot hence the expected value should be `pending-reboot`

**Comment by bryantbiggs** [comment]
the default by the AWS API is `static` so if you don't want that you need to follow the docs and set `pending-reboot` https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_parameter_group#apply_method

```hcl
  parameters = [
    {
      name         = "rds.logical_replication"
      value        = 1
      apply_method = "pending-reboot"
    }
  ]
```
